libs/utils/include/utils/Logger.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_UTILS_LOGGER_H
+#define TNT_UTILS_LOGGER_H
+
+// Logger.h provides a subset of the Abseil logging API, offering the following macros:
+
+// **LOG(severity)**: Logs a message at the specified severity level.
+// **DLOG(severity)**: Logs a message at the specified severity level only in debug builds.
+
+// Supported `severity` levels are:
+// * `INFO`
+// * `WARNING`
+// * `ERROR`
+// * `FATAL`
+
+// For programmatic control over logging severity, use the `LEVEL` macro:
+
+// **LOG(LEVEL(expression))**: Logs a message at a severity level determined by the `expression`.
+// The `expression` must return a `utils::LogSeverity` value, which is equivalent to
+// `absl::LogSeverity`.
+
+#if defined(FILAMENT_USE_ABSEIL_LOGGING)
+
+#include <absl/base/log_severity.h>
+#include <absl/log/log.h>
+
+namespace utils {
+using absl::LogSeverity;
+}
+
+#else
+
+#include <utils/Log.h>
+
+namespace utils {
+
+enum class LogSeverity : int {
+    kInfo = 0,
+    kWarning = 1,
+    kError = 2,
+    kFatal = 3,
+};
+
+template<typename Stream>
+class LogLine {
+public:
+    explicit LogLine(Stream& stream)
+        : mStream(stream) {}
+
+    LogLine(const LogLine&) = delete;
+    LogLine(LogLine&&) = delete;
+    LogLine& operator=(const LogLine&) = delete;
+    LogLine& operator=(LogLine&&) = delete;
+
+    ~LogLine() noexcept { mStream << utils::io::endl; }
```

Comment:
The semantics here is a little different from the old since it'll be flushed when the LogLine goes out of scope, as oppose to when `<<utils::io::endl` is used in the code.

Reply:
maybe that's ok?

Reply:
Yeah, the semantics are indeed different. I did my best to find all the places we _weren't_ using `utils::io::endl` and instead use a `stringstream` first to concatenate. IMO the way we did it explicitly with `utils::io::endl` is a better API, but this is just how Abseil works. I really wish Abseil had a `LOG_NOENDLINE`-style macro, but it doesn't.